src/transport/multicast/transport.c

```diff
@@ -206,6 +219,44 @@ void _z_multicast_transport_clear(_z_transport_t *zt) {
     _z_link_clear(&ztm->_link);
 }
 
+#if Z_FEATURE_MULTI_THREAD == 1
+z_result_t _z_multicast_tx_mutex_lock(_z_transport_multicast_t *ztm, bool block) {
+    if (block) {
+        _z_mutex_lock(&ztm->_mutex_tx);
+        return _Z_RES_OK;
+    } else {
+        return _z_mutex_try_lock(&ztm->_mutex_tx);
+    }
+}
+
+void _z_multicast_tx_mutex_unlock(_z_transport_multicast_t *ztm) { _z_mutex_unlock(&ztm->_mutex_tx); }
+
+void _z_multicast_rx_mutex_lock(_z_transport_multicast_t *ztm) { _z_mutex_lock(&ztm->_mutex_rx); }
+
+void _z_multicast_rx_mutex_unlock(_z_transport_multicast_t *ztm) { _z_mutex_unlock(&ztm->_mutex_rx); }
+
+void _z_multicast_peer_mutex_lock(_z_transport_multicast_t *ztm) { _z_mutex_lock(&ztm->_mutex_peer); }
+
+void _z_multicast_peer_mutex_unlock(_z_transport_multicast_t *ztm) { _z_mutex_unlock(&ztm->_mutex_peer); }
+
+#else
+z_result_t _z_multicast_tx_mutex_lock(_z_transport_multicast_t *ztm, bool block) {
+    _ZP_UNUSED(ztm);
+    _ZP_UNUSED(block);
+    return _Z_RES_OK;
+}
+void _z_multicast_tx_mutex_unlock(_z_transport_multicast_t *ztm) { _ZP_UNUSED(ztm); }
```

Comment:
Will compiler optimize empty method call, maybe will be better to use macros? They will be definitely optimized.
And even in multithreaded mode we can avoid unnecessary call. Other option will be use `inline` methods.